File: frontend/patient/app/src/main/java/com/pinghtdog/amping/api_schemas/Web_PatientDetailResponse.kt
// To parse the JSON, install kotlin's serialization plugin and do:
//
// val json                     = Json { allowStructuredMapKeys = true }
// val webPatientDetailResponse = json.parse(WebPatientDetailResponse.serializer(), jsonString)

package com.pinghtdog.amping.api_schemas

import kotlinx.serialization.*
import kotlinx.serialization.json.*
import kotlinx.serialization.descriptors.*
import kotlinx.serialization.encoding.*

@Serializable
data class WebPatientDetailResponse (
    val birthyear: Long,
    val contact: String,

    @SerialName("current_day")
    val currentDay: Long,

    val email: String,
    val firstname: String,
    val id: Long,
    val lastname: String,

    @SerialName("month3_protected")
    val month3Protected: Boolean,

    @SerialName("month_pdc")
    val monthPdc: Double,

    @SerialName("pdc_target")
    val pdcTarget: Double,

    @SerialName("regimen_start")
    val regimenStart: String,

    @SerialName("total_days")
    val totalDays: Long
)

File: frontend/hc_professional/src/api_types/Web_PatientDetailResponse.ts
export interface WebPatientDetailResponse {
    birthyear:        number;
    contact:          string;
    current_day:      number;
    email:            string;
    firstname:        string;
    guardians:        WebPatientGuardianDetailEntry[];
    id:               number;
    lastname:         string;
    month3_protected: boolean;
    month_pdc:        number;
    pdc_target:       number;
    regimen_start:    Date;
    total_days:       number;
}

export interface WebPatientGuardianDetailEntry {
    contact:   string;
    email:     string;
    firstname: string;
    id:        number;
    lastname:  string;
}
